llvm/test/DebugInfo/Generic/is_stmt-at-block-start.ll

```diff
@@ -0,0 +1,38 @@
+;; Checks that when an instruction at the start of a BasicBlock has the same
+;; DebugLoc as the instruction at the end of the previous BasicBlock, we add
+;; is_stmt to the new line, to ensure that we still step on it if we arrive from
+;; a BasicBlock other than the immediately preceding one.
+
+; RUN: %llc_dwarf -O0 -filetype=obj < %s | llvm-dwarfdump --debug-line - | FileCheck %s
+
+; CHECK:      {{0x[0-9a-f]+}}     13      5 {{.+}} is_stmt
+; CHECK-NEXT: {{0x[0-9a-f]+}}     13     25 {{.+}} is_stmt
```

Comment:
ISTM the first check is for block if.then2, the second is for if.else4? Should there also be one for if.end8?

Reply:
Not for this test specifically, as only these two instructions have non-0 line numbers (excluding 0x0, which has the function header).

Reply:
Oh right, `!dbg` isn't inherited the way it is for asm.
But, you could use the same `!dbg` for both, showing that it really is the BB that triggers it.

Reply:
Hm, trying to make this work has revealed another an issue - when the lines are actually identical, they will be merged into a single entry (whether this patch is in effect or not), which any jump to the branch would land in the middle of. I'm not sure whether that would result in a step in the debugger or not; it prevents the test from working, at the very least.